### PR TITLE
Testing the ECC circuits

### PR DESCRIPTION
**Pull request by vb7401**
Issue #7 finished:
- reviewed all circuit code closely and commented to test understanding + help future devs
- reviewed test cases and added some notes on the current testing strategy
- Ran Ecne on all the functions without Keccak (those took too long and would likely fail), results here: https://github.com/vb7401/EcneProject/blob/master/ecne_results.txt
  - I think it's expected that most of the ECC functions have "potentially unsound constraints" as we are omiting the check that these points are actually on the curve and the formulas assume this. Good that Poseidon + PubKeyMembership have sound constraints however!

### CODE REVIEW

**Review by DanTehrani — APPROVED** [review]
@vb7401  These comments are really well-written! I'll try to include similar comments when writing future code.
Ecne checks are really cool, glad to know there are no obvious red flags.

LGTM!

File: packages/circuits/eff_ecdsa_membership/eff_ecdsa.circom [hunk 1]
```diff
@@ -15,17 +21,13 @@ template EfficientECDSA() {
     signal output pubKeyX;
     signal output pubKeyY;
 
-    var gX = 55066263022277343669578718895168534326250603453777594175500187360389116729240;
```
Comment (DanTehrani) [inline packages/circuits/eff_ecdsa_membership/eff_ecdsa.circom:18]: Updated the hosted circuit files to reflect this change

**Comment by lsankar4033** [comment]
@vb7401 I wonder what the right future policy is for ecne-like verification of things like curve math (and the omission of constraint that points on curve). I suppose that 'philosophically' the constraint is baked into the naming of the circuits so maybe just not worth worrying about

File: packages/circuits/eff_ecdsa_membership/secp256k1/add.circom [hunk 2]
```diff
@@ -3,6 +3,16 @@ pragma circom 2.1.2;
...
+/**
+ *  Secp256k1AddIncomplete
+ *  ======================
+ *
+ *  Adds two points (xP, yP) and (xQ, yQ) on the secp256k1 curve. This function 
+ *  only works for points where xP != xQ and are not at infinity. We can implement 
+ *  the raw formulae for this operation as we are doing right field arithmetic 
+ *  (we are doing secp256k1 base field arithmetic in the secq256k1 scalar field, 
+ *  which are equal). Should work for any short Weierstrass curve (Pasta, P-256).
+ */
```
Comment (lsankar4033) [inline packages/circuits/eff_ecdsa_membership/secp256k1/add.circom:15]: ah ya, good call documenting these two

**Pull request merged**